meteor/sofie-server/data-access/repositories/test/mongo-rundown-repository.spec.ts

```diff
@@ -117,13 +118,88 @@ describe(`${MongoRundownRepository.name}`, () => {
 		})
 	})
 
+	describe(`${MongoRundownRepository.prototype.saveRundown.name}`, () => {
+		it('has rundown as not on air and saves the rundown as on air', async () => {
+			const id: string = 'rundownId'
+			const inactiveRundown: MongoRundown = createMongoRundown({
+				_id: id,
+			})
+			const activeRundown: Rundown = createRundown({ rundownId: id, isRundownActive: true })
+
+			await testDatabase.populateDatabaseWithInactiveRundowns([inactiveRundown])
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const mongoDb: MongoDatabase = mock(MongoDatabase)
+			const db: Db = testDatabase.getDatabase()
+			const collection = db.collection(COLLECTION_NAME)
+
+			when(mongoDb.getCollection(anything())).thenReturn(collection)
+			when(mongoConverter.convertToMongoRundown(anything())).thenReturn({
+				_id: activeRundown.id,
+				name: activeRundown.name,
+				isActive: activeRundown.isActive(),
+			} as unknown as MongoRundown)
+
+			const testee: RundownRepository = createTestee({
+				mongoDb: mongoDb,
+				mongoConverter: mongoConverter,
+			})
+			await testee.saveRundown(activeRundown)
+
+			const result: MongoRundown = (await db
+				.collection(COLLECTION_NAME)
+				.findOne({ _id: id })) as unknown as MongoRundown
+
+			expect(result.isActive).toBeTruthy()
+		})
+
+		it('has rundown as on air and saves the rundown as not on air', async () => {
+			const id: string = 'rundownId'
+			const activeRundown: MongoRundown = createMongoRundown({ _id: id })
+			const inactiveRundown: Rundown = createRundown({ rundownId: id, isRundownActive: false })
+
+			await testDatabase.populateDatabaseWithActiveRundowns([activeRundown])
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const mongoDb: MongoDatabase = mock(MongoDatabase)
+			const db: Db = testDatabase.getDatabase()
+			const collection = db.collection(COLLECTION_NAME)
+
+			when(mongoDb.getCollection(anything())).thenReturn(collection)
+			when(mongoConverter.convertToMongoRundown(anything())).thenReturn({
+				_id: inactiveRundown.id,
+				name: inactiveRundown.name,
+				isActive: inactiveRundown.isActive(),
+			} as unknown as MongoRundown)
+
+			const testee: RundownRepository = createTestee({
+				mongoDb: mongoDb,
+				mongoConverter: mongoConverter,
+			})
+			await testee.saveRundown(inactiveRundown)
+
+			const result: MongoRundown = (await db
+				.collection(COLLECTION_NAME)
+				.findOne({ _id: id })) as unknown as MongoRundown
+
+			expect(result.isActive).toBeFalsy()
+		})
+	})
+
+	// TODO: Extract to Helper Class in Model layer
 	function createMongoRundown(mongoRundownInterface?: Partial<MongoRundown>): MongoRundown {
```

Comment:
Comment on the TODO:
This creates a `MongoRundown` which is database specific knowlegde and should therefore _not_ leave the DataAccessLayer.

The `createRundown` method just below however should be replaced by the new `EntityMockFactory`

Reply:
This is a leftover from the "Delete-PR" i didn't make this, i don't know why that is there. I'll remove the TODO's.

Reply:
Has been removed!